openhcl/virt_mshv_vtl/src/processor/snp/mod.rs

```diff
@@ -1441,35 +1443,28 @@ impl UhProcessor<'_, SnpBacked> {
 impl<T: CpuIo> X86EmulatorSupport for UhEmulationState<'_, '_, T, SnpBacked> {
     type Error = UhRunVpError;
 
-    fn vp_index(&self) -> VpIndex {
-        self.vp.vp_index()
-    }
-
-    fn vendor(&self) -> x86defs::cpuid::Vendor {
-        self.vp.partition.caps.vendor
-    }
-
-    fn state(&mut self) -> Result<x86emu::CpuState, Self::Error> {
+    fn load_registers(&mut self) {
         let vmsa = self.vp.runner.vmsa(self.vtl);
-        Ok(x86emu::CpuState {
-            gps: [
-                vmsa.rax(),
-                vmsa.rcx(),
-                vmsa.rdx(),
-                vmsa.rbx(),
-                vmsa.rsp(),
-                vmsa.rbp(),
-                vmsa.rsi(),
-                vmsa.rdi(),
-                vmsa.r8(),
-                vmsa.r9(),
-                vmsa.r10(),
-                vmsa.r11(),
-                vmsa.r12(),
-                vmsa.r13(),
-                vmsa.r14(),
-                vmsa.r15(),
-            ],
+        let gps = [
```

Comment:
If literally everything is in the vmsa then we can probably just not use a cache for SNP (i.e. have the cache type be ()) and just read and modify directly. Touching the vmsa is cheap.

Reply:
Makes sense, done.